### PR TITLE
Make nodejs_waiter aware of pending validation

### PR DESCRIPTION
**Pull request by maackle**
Let's see if this helps with the hanging tests.
- Adds a wait condition: Every AddPendingValidation must be matched with a RemovePendingValidation
- DRY's waiter logic a bit 

- [ ] I have added a summary of my changes to the changelog

### CODE REVIEW

File: nodejs_waiter/src/waiter.rs [hunk 1]
```diff
@@ -4,7 +4,7 @@ use holochain_core::{
     nucleus::ZomeFnCall,
     signal::{Signal, SignalReceiver},
 };
-use holochain_core_types::entry::Entry;
+use holochain_core_types::{cas::content::AddressableContent, entry::Entry};
```
Comment (Connoropolous) [inline nodejs_waiter/src/waiter.rs]: why does this come in?

Comment (zippy) [reply nodejs_waiter/src/waiter.rs]: If you use `.address()` on something that implements AddressableContent, you also have to include the trait in scope.

Comment (Connoropolous) [reply nodejs_waiter/src/waiter.rs]: I see

**Comment by zippy** [comment]
@pjkundert can you see if this fixes some of the hanging problems you were having in your tests?

File: nodejs_waiter/src/waiter.rs (hunk 1 above)
Comment (maackle) [reply nodejs_waiter/src/waiter.rs]: Yeah here https://github.com/holochain/holochain-rust/pull/1072/files/8ddb359fbe20ed4521311e00b3d51376c835fbcc#diff-f6e488476c3a39ed86c9080bee61ef9aR209

**Comment by maackle** [comment]
@Connoropolous there are tests: https://github.com/holochain/holochain-rust/blob/develop/nodejs_waiter/src/waiter.rs#L433

**Comment by pjkundert** [comment]
Sorry, yes -- I tested this.  No effect.

File: nodejs_conductor/publish.js [hunk 2]
```diff
@@ -58,7 +64,7 @@ npmPackageJson.scripts.fallback = "npm run compile && mkdir -p bin-package && cp
 
 fs.writeFileSync("./dist/package.json", JSON.stringify(npmPackageJson, null, 2));
 
-shell.mkdir("./bin-package");
+shell.mkdir('-p', "./bin-package");
```
Comment (Connoropolous) [inline nodejs_conductor/publish.js]: what's -p?

**Comment by Connoropolous** [comment]
Cool, good to know there's some tests, this feels like a pretty complex area in terms of functionality, test coverage here is a big deal I think

File: nodejs_conductor/publish.js (hunk 2 above)
Comment (maackle) [reply nodejs_conductor/publish.js]: recursive -- the easiest way to make it not complain if the dir already exists, but still create it if it doesn't

**Comment by zippy** [comment]
Even though this doesn't fix @pjkundert blocking issue, I think this is ready to merge no?  We still need it.

**Pull request merged**